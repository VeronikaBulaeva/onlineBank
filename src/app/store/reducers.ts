import { combineReducers } from "@reduxjs/toolkit";
import creditSlice from "./slices/creditSlice.ts";

const rootReducer = combineReducers({
  credit: creditSlice,
});

export default rootReducer;
