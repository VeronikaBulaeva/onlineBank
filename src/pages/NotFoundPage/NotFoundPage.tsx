import { FC } from "react";
import styles from "./notFoundPage.module.css";
import DefaultButton from "@/components/shared/DefaultButton/DefaultButton.tsx";
import { ButtonType } from "@/components/shared/DefaultButton/types.ts";
import ErrorImage from "@/assets/error404.svg";
import { HOME_PAGE_ROUTE } from "@/constants/routes.ts";

const NotFoundPage: FC = () => {
  return (
    <section className={styles.errorPage}>
      <div className={styles.textBlock}>
        <p className={styles.textBlock__title}>Oops....</p>
        <p className={styles.textBlock__text}>Page not found</p>
        <p className={styles.textBlock__description}>
          This Page doesn`t exist or was removed! We suggest you go back.
        </p>
        <DefaultButton
          buttonType={ButtonType.link}
          className={styles.button}
          link={HOME_PAGE_ROUTE}
        >
          <p>Go back</p>
        </DefaultButton>
      </div>
      <div>
        <img src={ErrorImage} alt="error" />
      </div>
    </section>
  );
};

export default NotFoundPage;
