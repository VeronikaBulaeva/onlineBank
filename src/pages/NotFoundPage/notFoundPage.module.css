.errorPage {
    max-width: 1300px;
    margin-inline: auto;
    padding-block: 20px 98px;
    display: flex;
    justify-content: space-between;
}

.textBlock {
    display: grid;
    gap: 32px;
    justify-items: flex-start;
    align-content: center;
}

.textBlock__title {
    font-weight: 700;
}

.textBlock__text, .textBlock__description {
    font-weight: 500;
}

.textBlock__title, .textBlock__text {
    font-family: "Ubuntu", sans-serif;
    font-size: 30px;
    line-height: 32px;
    color: rgba(28, 28, 30, 1);
}

.textBlock__description {
    font-size: 16px;
    font-weight: 500;
    line-height: 18px;
    color: rgba(75, 75, 75, 1);
}

.button {
    width: 148px;
    height: 48px;
    max-width: none;
}

@media screen and (max-width: 1300px) {
    .errorPage {
        padding-inline: 8px;
    }
}

@media screen and (max-width: 660px) {
    .button {
        width: 100%;
    }
}
