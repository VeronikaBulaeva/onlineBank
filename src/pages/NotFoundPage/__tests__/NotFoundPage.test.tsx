import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "@/app/store/store.ts";
import { BrowserRouter as Router } from "react-router-dom";
import NotFoundPage from "../NotFoundPage.tsx";

describe("LoanPage", () => {
  it("renders correctly", async () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <NotFoundPage />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
