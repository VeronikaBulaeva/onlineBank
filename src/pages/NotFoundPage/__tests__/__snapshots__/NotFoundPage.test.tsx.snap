// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoanPage > renders correctly 1`] = `
<div>
  <section
    class="_errorPage_9c8518"
  >
    <div
      class="_textBlock_9c8518"
    >
      <p
        class="_textBlock__title_9c8518"
      >
        Oops....
      </p>
      <p
        class="_textBlock__text_9c8518"
      >
        Page not found
      </p>
      <p
        class="_textBlock__description_9c8518"
      >
        This Page doesn\`t exist or was removed! We suggest you go back.
      </p>
      <a
        class="_button_148529 _button_9c8518"
        href="/"
      >
        <p>
          Go back
        </p>
      </a>
    </div>
    <div>
      <img
        alt="error"
        src="/src/assets/error404.svg"
      />
    </div>
  </section>
</div>
`;
